### ISSUE TITLE
'Initialize a Repository' on folder not owned

### ISSUE THREAD

**Issue by JR-Bae** [jupyterlab/jupyterlab-git#1141]
## Description

In jupyter lab, click Git -> Initialize a Repository. It could not initiate repository in some directory. It could not browse repository.
 
## Reproduce

![git1](https://user-images.githubusercontent.com/25498289/177695788-4b63289b-6e3c-460e-a633-8f2b2ee97322.gif)
![git2](https://user-images.githubusercontent.com/25498289/177695812-f4bf7db1-0141-4a4f-b1fa-ed7f6368ef18.gif)

## Expected behavior
Now I can browse my repository after running 'git config --global --add safe.directory /home/work'.

git version upgrade 2.25.1 to 2.35.2 in jupyterlab-git.
ref: https://github.blog/2022-04-12-git-security-vulnerability-announced/

So I recommend clicking on 'Initialize Repository' which will bring up a popup menu where you choose whether to set up a safe directory or not.

## Context

<img width="785" alt="image" src="https://user-images.githubusercontent.com/25498289/177694072-77d0a177-7572-454c-abe2-b9f30193158e.png">

- Python package version:
Name: jupyterlab-git
Version: 0.37.1
Summary: A JupyterLab extension for version control using git
Home-page: https://github.com/jupyterlab/jupyterlab-git
Author: Jupyter Development Team
Author-email: 
License: BSD-3-Clause
Location: /opt/conda/lib/python3.7/site-packages
Requires: jupyter-server, nbdime, nbformat, packaging, pexpect
Required-by: 

- Extension version:
JupyterLab v3.4.3
/opt/conda/share/jupyter/labextensions
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
        jupyterlab-execute-time v2.1.0 enabled OK (python, jupyterlab_execute_time)
        jupyterlab-system-monitor v0.8.0 enabled OK (python, jupyterlab-system-monitor)
        jupyterlab-topbar-extension v0.6.1 enabled OK (python, jupyterlab-topbar)
        nbdime-jupyterlab v2.1.1 enabled OK
        ipyparallel-labextension v8.4.1 enabled OK
        jupyter-matplotlib v0.11.1 enabled OK
        @jupyter-server/resource-usage v0.6.1 enabled OK (python, jupyter-resource-usage)
        @jupyterlab/git v0.37.1 enabled OK (python, jupyterlab-git)
        @krassowski/jupyterlab-lsp v3.10.1 enabled OK (python, jupyterlab-lsp)
        @jupyter-widgets/jupyterlab-manager v3.1.1 enabled OK (python, jupyterlab_widgets)

Other labextensions (built into JupyterLab)
   app dir: /opt/conda/share/jupyter/lab

- Git version:
git version 2.25.1

- Operating System and its version:
Ubuntu 20.04.2 LTS

## Web Browser Output
<pre>
{
    "code": 128,
    "command": "git rev-parse --show-prefix",
    "message": "fatal: unsafe repository ('/home/work' is owned by someone else)\nTo add an exception for this directory, call:\n\n\tgit config --global --add safe.directory /home/work\n"
}
</pre>

**Comment by JR-Bae** [comment]
@fcollonval It is running using a container k8s pod but initiating a repository could be run on jupyterlab v3.3.0 before jupyterlab upgrade to v3.4.3.
Upgrade git version 2.25.1 to 2.37.0. Now it is good running.

**Comment by fcollonval** [comment]
> Does upgrading 'jupyterlab-git' automatically upgrade 'native git'?

No as the extension is a Python package there is no easy way to install a tool like git with it.

**Comment by fcollonval** [comment]
Closing as won't fix
